File: Python/student.py
from person import Person

# Implemented Inheritance

class Student(Person):

    def __init__(self, id, name, address, phone):
        super().__init__(id, name)
        self.__address = address
        self.__phone = phone

    def getAddress(self):
        return self.__address

    def getPhone(self):
        return self.__phone

    def setAddress(self, address):
        self.__address = address

    def setPhone(self, phone):
        self.__phone = phone

    def showRole(self):
        print(self.getName() + " is a Student")

    def enroll(self, p):
        print(self.getName() + " enrolled in " + p.name)

File: C#/Student.cs
using System;

namespace StudentManagementSystem
{
    // Implemented Inheritance
    public class Student : Person
    {
        private string address;
        private string phone;

        public Student(int id, string name, string address, string phone)
            : base(id, name)
        {
            this.address = address;
            this.phone = phone;
        }

        public string Address
        {
            get { return address; }
            set { address = value; }
        }

        public string Phone
        {
            get { return phone; }
            set { phone = value; }
        }

        public override void ShowRole()
        {
            Console.WriteLine(Name + " is a Student");
        }

        public void Enroll(Program p)
        {
            Console.WriteLine(Name + " enrolled in " + p.name);
        }
    }
}
